libraries/rib-base/src/main/java/com/badoo/ribs/android/dialog/DialogExtensions.kt

```diff
@@ -45,8 +46,14 @@ private fun AlertDialog.Builder.setRib(dialog: Dialog<*>, context: Context) {
         setView(object : FrameLayout(context) {
             val host = AndroidRibViewHost(this)
 
-            override fun onAttachedToWindow() {
-                super.onAttachedToWindow()
+            init {
+                // If we will execute multiple configurations in a short period of time ->
+                // Node can be attached to a host multiple times ->
+                // java.lang.IllegalStateException: The specified child already has a parent.
+                // Detach node here -> no harmful effect when view is not attached
+                // PushTwoPopOneDefaultTest.noPermanent_singleInitial_pushOverlay_pushContent_pop - to reproduce
+                host.detachChild(it.node)
```

Comment:
If this is needed to make it pass, the problem is probably somewhere else, and should be fixed there?

Reply:
I am not sure, in theory it can also happen in production environment.
Here are more details on why it's happening

```
[configurationDialog] push
AlertDialog.show()
[configuration] push
AlertDialog.dismiss() -> Here we starting dismissing the dialog after new configuration
[configuration] pop -> But we immediately popping new config
AlertDialog.show() -> And show dialog again without detaching the view from host
onDismissListener
AlertDialog.dismiss()
onDismissListener
```